doc/articles/features/open-id-connect.md

```diff
@@ -0,0 +1,206 @@
+
+**WebAssembly**
+
+There's nothing special for UWP. The default _return uri_ of the platform (`WebAuthenticationBroker.GetCurrentApplicationCallbackUri()`) will work with this sample and will default to `<origin>/authentication-callback`. It should be something like `http://localhost:5000/authentication-callback` when running locally using Kestrel.
+
+## Step 4 - Prepare the UI
+
+Add the following button in your application:
+
+``` xml
+<StackPanel Orientation="Horizontal" Spacing="5">
+	<Button Click="SignIn_Clicked" x:Name="btnSignin" IsEnabled="False">Sign In</Button>
+    <Button Click="SignOut_Clicked" x:Name="btnSignout" IsEnabled="False">Sign Out</Button>
+</StackPanel>
+```
+
+## Step 5 - Prepare the requesting code
+
+Add the following code to the main page of your application:
+
+``` csharp
+// Put this code in class of the MainPage.xaml.cs
+
+public MainPage()
+{
+	this.InitializeComponent();
+	PrepareClient();
+}
+
+private OidcClient _oidcClient;
+private AuthorizeState _loginState;
+private Uri _logoutUrl;
+
+private async void PrepareClient()
+{
+	var redirectUri = WebAuthenticationBroker.GetCurrentApplicationCallbackUri().OriginalString;
+
+	// Create options for endpoint discovery
+	var options = new OidcClientOptions
+	{
+		Authority = "https://demo.identityserver.io",
+		ClientId = "interactive.confidential",
+		ClientSecret = "secret",
+		Scope = "openid profile email api offline_access",
+		RedirectUri = redirectUri,
+		PostLogoutRedirectUri = redirectUri,
+		ResponseMode = OidcClientOptions.AuthorizeResponseMode.Redirect,
+		Flow = OidcClientOptions.AuthenticationFlow.AuthorizationCode
+	};
+
+    // Create the client. In production application, this is often created and stored
+    // directly in the Application class.
+	_oidcClient = new OidcClient(options);
+
+    // Invoke Discovery and prepare a request state, containing the nonce.
+    // This is done here to ensure the discovery mecanism is done before
+    // the user clicks on the SignIn button. Since the opening of a web window
+    // should be done during the handling of a user interaction (here it's the button click),
+    // it will be too late to reach the discovery endpoint.
+    // Not doing this could trigger popup blockers mechanisms in browsers.
+	_loginState = await _oidcClient.PrepareLoginAsync();
+    btnSignin.IsEnabled = true;
+
+    // Same for logout url.
+    _logoutUrl = new Uri(await _oidcClient.PrepareLogoutAsync(new LogoutRequest()));
+    btnSignout.IsEnabled = true;
+}
+```
+
+## Step 6 - Proceed to authentication
+
+Add following button handlers:
+
+``` csharp
+private async void SignIn_Clicked(object sender, RoutedEventArgs e)
+{
+    var startUri = new Uri(_loginState.StartUrl);
+
+	// Important: there should be NO await before calling .AuthenticateAsync() - at least
+	// on WebAssembly, in order to prevent trigering the popup blocker mechanisms.
```

Comment:
```suggestion
	// Important: there should be NO await before calling .AuthenticateAsync() - at least
	// on WebAssembly, in order to avoid triggering the popup blocker mechanisms.
```